### ISSUE TITLE
Unable to embed badge using the <object> tag

### ISSUE THREAD

**Issue by arunisaac** [repology/repology-webapp#175]
Hi,

I wish to embed a repology badge using the <object> tag instead of the <img> tag. But, when I do so, I get a message saying "To protect your security, repology.org will not allow IceCat to display the page if another site has embedded it. To see this page, you need to open it in a new window". Is this perhaps due to the content-security-policy header set by repology-webapp? Can that be changed to allow embedding using the <object> tag?

I'm using Emacs Org mode to generate my HTML. Org mode uses <object> tags for SVG images. Hence, it is not easy for me to switch to the <img> tag.

Thanks!

**Comment by AMDmi3** [comment]
I'd prefer to avoid introducing quirks for the sake of broken clients. Besides, I have no idea of how this can be fixed. In theory it is controlled by `Access-Control-Allow-Origin` header, but I couldn't manage to make firefox display a badge as `<object>` with any settings.

`access-control-allow-origin: *` is currently enabled, does it work for you?

**Comment by arunisaac** [comment]
I experimented with a local server serving a badge. It seems like the
X-Frame-Options and Content-Security-Policy headers are the
problem. When both do not exist, the badge embeds fine. I checked with
both IceCat and Firefox. Specifically, in the Content-Security-Policy
header, the "frame-ancestors 'none'" directive is the
problem. "default-src 'none'" might also be problematic, but at least in
my experiments, the badge embedded correctly even with it.

See
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options
and
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy
for details about the headers.
